crypto/ocsp/ocsp_http.c

```diff
@@ -176,12 +177,20 @@ int OCSP_REQ_CTX_nbio(OCSP_REQ_CTX *rctx) {
 
       OPENSSL_FALLTHROUGH;
     case OHS_ASN1_WRITE_INIT:
-      rctx->asn1_len = BIO_get_mem_data(rctx->mem, NULL);
+      if(!BIO_mem_contents(rctx->mem, NULL, &tmp_data_len)) {
+        rctx->state = OHS_ERROR;
+        return 0;
+      }
+      rctx->asn1_len = (unsigned long)tmp_data_len;
       rctx->state = OHS_ASN1_WRITE;
 
       OPENSSL_FALLTHROUGH;
     case OHS_ASN1_WRITE:
-      data_len = BIO_get_mem_data(rctx->mem, &data);
+      if(!BIO_mem_contents(rctx->mem, &data, &tmp_data_len)) {
+        rctx->state = OHS_ERROR;
+        return 0;
+      }
+      data_len = (int)tmp_data_len;
```

Comment:
For all of these int casts in the incredibly unlikely event that the bio holds INT_MAX + 1 bytes this will underflow and could do something weird.